test/freenet/client/filter/M3UFilterTest.java

```diff
@@ -35,16 +34,14 @@ public void testSuiteTest() throws IOException {
             try {
                 ibo = resourceToBucket(original);
             } catch (IOException e) {
-                System.out.println(original + " not found, test skipped");
-                unreadableFiles.add(original);
-                continue;
+                System.out.println(original + " not found");
```

Comment:
If you remove the whole try-catch block here you can also get rid of parts of the `resourceToBucket` method. Crash & burn! \o/

Reply:
I want the filename in the error message — that’s the whole reason for the try-catch block.

Reply:
Why?

* If this test fails during `gradle test` with an exception you run it exactly once in the IDE to find out why it’s failing. The filename in the exception will not yield any meaningful information (in the sense that it saves you a lot of time).
* You could add the filename to the exception you create in the `resourceToBucket` method, this way it will show up without the try-catch block.

Reply:
you’re right — resourceToBucket now preserves the filename and this makes the code much nicer. Thank you!